cairo/tests/utils/message_builder.py

```diff
@@ -0,0 +1,108 @@
+from ethereum_types.numeric import U256, Uint
+from hypothesis import strategies as st
+
+from ethereum.cancun.fork_types import Address
+from tests.utils.args_gen import Message
+from tests.utils.strategies import (
+    accessed_addresses,
+    accessed_storage_keys,
+    address_zero,
+    code,
+    gas_left,
+    small_bytes,
+    uint,
+)
+
+
+class MessageBuilder:
+
+    def __init__(self):
+        self._caller = st.just(address_zero)
+        self._target = st.just(address_zero)
+        self._current_target = st.just(address_zero)
+        self._gas = st.just(Uint(0))
+        self._value = st.just(U256(0))
+        self._data = st.just(b"")
+        self._code_address = st.none()
+        self._code = st.just(b"")
+        self._depth = st.just(Uint(0))
+        self._should_transfer_value = st.just(False)
+        self._is_static = st.just(False)
+        self._accessed_addresses = st.just(set())
+        self._accessed_storage_keys = st.just(set())
+        self._parent_evm = st.none()
+
+    def with_caller(self, strategy=st.from_type(Address)):
+        self._caller = strategy
+        return self
+
+    def with_target(self, strategy=st.from_type(Address)):
+        self._target = strategy
+        return self
+
+    def with_current_target(self, strategy=st.from_type(Address)):
+        self._current_target = strategy
+        return self
+
+    def with_gas(self, strategy=gas_left):
+        self._gas = strategy
+        return self
+
+    def with_value(self, strategy=st.from_type(U256)):
+        self._value = strategy
+        return self
+
+    def with_data(self, strategy=small_bytes):
+        self._data = strategy
+        return self
+
+    def with_code_address(self, strategy=st.from_type(Address) | st.none()):
```

Comment:
nit, but if the type of code address is Optional[Address] in the spec we can reuse it as is 
```suggestion
    def with_code_address(self, strategy=st.from_type(Optional[Address])):
```